org.eclipse.buildship.core.test/src/main/groovy/org/eclipse/buildship/core/proxy/ProxySettingsTest.groovy

```diff
@@ -0,0 +1,348 @@
+        File rootProject = newProject(false, true)
+        server.expectGet("/not-a-real-group/not-a-real-dependency/0.0/not-a-real-dependency-0.0.pom", new File(dependencyTempFolder.root, 'not-a-real-dependency-0.0.pom'))
+        def job = new RunGradleConfigurationDelegateJob(createLaunchMock(), createLaunchConfigurationMock(rootProject.absolutePath))
+
+        when:
+        job.schedule()
+        job.join()
+
+        then:
+        proxyServer.requestCount == 1
+    }
+
+    def "Proxies are not accessed from Gradle build VM with incorrect proxy settings"() {
+        setup:
+        proxyServer.start()
+        proxyServer.requireAuthentication(userId, password)
+        setupTestProxyData("localhost", proxyServer.port, userId, password + '-incorrect-password')
+        File rootProject = newProject(false, true)
+        def job = new RunGradleConfigurationDelegateJob(createLaunchMock(), createLaunchConfigurationMock(rootProject.absolutePath))
+
+        when:
+        job.schedule()
+        job.join()
+
+        then:
+        proxyServer.requestCount == 0
+    }
+
+    def setupTestProxyData(String host, int port, String userId, String password) {
+        String proxyHostMemory = System.getProperty("http.proxyHost")
+        String proxyPortMemory = System.getProperty("http.proxyPort")
+        String proxyUserMemory = System.getProperty("http.proxyUser")
+        String proxyPassMemory = System.getProperty("http.proxyPassword")
+
+
+        // Gets rid of The HTTP proxy is enabled in the preferences but disabled in the system settings
+        final IProxyData httpProxyData = proxyService.getProxyData(IProxyData.HTTP_PROXY_TYPE)
+
+        httpProxyData.setHost(host)
+        httpProxyData.setPort(port)
+        httpProxyData.setUserid(userId)
+        httpProxyData.setPassword(password)
+        proxyService.setProxyData((IProxyData[]) [
+            httpProxyData
+        ])
+
+        if (proxyHostMemory != null) {
+            System.setProperty("http.proxyHost", proxyHostMemory)
+        }
+        if (proxyPortMemory != null) {
+            System.setProperty("http.proxyPort", proxyPortMemory)
+        }
+        if (proxyUserMemory != null) {
+            System.setProperty("http.proxyUser", proxyUserMemory)
+        }
+        if (proxyPassMemory != null) {
+            System.setProperty("http.proxyPassword", proxyPassMemory)
+        }
+
+    }
+
+    def newProjectImportJob(File location) {
+        ProjectImportConfiguration configuration = new ProjectImportConfiguration()
+        configuration.gradleDistribution = GradleDistributionWrapper.from(GradleDistribution.forRemoteDistribution("http://not.a.real.domain/gradlew/dist".toURI()))
+
+        // Ian-todo: In PR, emphasize that gradleUserHome has been changed. Perhaps also test to see if gradle distribution exists in workspace.
+        configuration.gradleUserHome = new File(ResourcesPlugin.getWorkspace().root.rawLocation.toString())
```

Comment:
The ProjectImportSpecification is not complete in the sense of parametrization. You can define a new `newProjectPreviewJob()` method there. I'd suggest the following signature: `newProjectPreviewJob(File, ProjectImportConfiguration)`.